common/djangoapps/student/views.py

```diff
@@ -426,9 +426,9 @@ def _cert_info(user, course_overview, cert_status, course_mode):  # pylint: disa
         persisted_grade = CourseGradeFactory().read(user, course=course_overview)
         if persisted_grade is not None:
             status_dict['grade'] = unicode(persisted_grade.percent)
-        elif 'grade' in cert_status:
-            status_dict['grade'] = cert_status['grade']
-        else:
+        if 'grade' not in status_dict or int(cert_status.get('grade', '0')) > persisted_grade.percent:
```

Comment:
What if `persisted_grade` is `None` on this line?

I think this block will be cleaner if we explicitly get the 2 possible grades and compare them, as opposed to the sequential construction.
```
cert_grade = -1
robust_grade = -1
if persisted_grade is not None:
    robust_grade = unicode(persisted_grade.percent)
if 'grade' in cert_status:
    cert_grade = cert_status['grade']

if cert_grade == -1 and robust_grade == -1:
    return default_info

status_dict['grade'] = max(cert_grade, robust_grade)
```